core/commands/pin.go

```diff
@@ -51,13 +57,26 @@ on disk.
 			recursive = false
 		}
 
-		_, err = pin(n, req.Arguments(), recursive)
+		added, err := corerepo.Pin(n, req.Arguments(), recursive)
 		if err != nil {
 			return nil, err
 		}
 
-		// TODO: create some output to show what got pinned
-		return nil, nil
+		return &PinOutput{added}, nil
+	},
+	Marshalers: cmds.MarshalerMap{
+		cmds.Text: func(res cmds.Response) (io.Reader, error) {
+			added, ok := res.Output().(*PinOutput)
+			if !ok {
+				return nil, u.ErrCast()
+			}
+
+			buf := new(bytes.Buffer)
+			for _, k := range added.Pinned {
+				fmt.Fprintf(buf, "Pinned %s\n", k)
```

Comment:
lowercase "pinned". it should probably say the type?

```
pinned %s recursively
pinned %s directly
```
(this order makes sense grammatically, and also makes the output all aligned consistently, for easy scanning of many lines of output)